manifests/service.pp

```diff
@@ -5,14 +5,28 @@
 ){
   require zookeeper::install
 
-  service { 'zookeeper':
-    ensure     => 'running',
-    hasstatus  => true,
-    hasrestart => true,
-    enable     => true,
-    require    => [
-      Package['zookeeperd'],
-      File["${cfg_dir}/zoo.cfg"]
-    ]
+  if $::lsbdistcodename == 'xenial' {
```

Comment:
I don't think you need this at all, you should be able to get away with just "zookeeper" and let puppet autodetect the provider.

Reply:
You can do that, or make the `package[zookeeperd]` have a `before`.

Reply:
Updated/fixed this.  service.pp is not altered now except  the before clause in install.pp. and removal of package[zookeeperd] from service.pp.